### DISCUSSION TITLE
How to change keyboard variant?

### DISCUSSION THREAD

**Discussion by Kazuto** [hyprwm/Hyprland#2372] [Q&A]
For hours now I'm trying to change my keyboard variant but I just can't get it to work.

I've got a Logitech MX Master Keys with mac keys because I work with a mac in my job.

Running `localectl list-keymaps | grep de` I can see a record `mac-de-latin1-nodeadkeys` but no matter what I tried I can't get it to work because the documentation is rather poorly written.

What I've tried so far in my hyprland.conf:

```
input {
  kb_layout = de
  kb_variant = ,mac
  follow_mouse = 1
  sensitivity = 0 # -1.0 - 1.0, 0 means no modification.
}
```

```
input {
  kb_layout = de
  kb_variant = mac,latin1,nodeadkeys
  follow_mouse = 1
  sensitivity = 0 # -1.0 - 1.0, 0 means no modification.
}
```

**Comment by 11xx** [comment]
I think you're missing the match between kb_layout and kb_variant, explained in the [switchxkblayout note](https://wiki.hyprland.org/Configuring/Using-hyprctl/#:~:text=however%20the%20amount%20of%20comma-separated%20parameters%20have%20to%20match):

so what you would want is:
```
  kb_layout = de,de # repeat to match the number of variants using only one layout
  kb_variant = ,mac
```
or
```
  kb_layout = de,de,de
  kb_variant = mac,latin1,nodeadkeys
```

then switch with `hyprctl switchxkblayout keyboard-device-name next`

**Reply by Kazuto** [reply]
Using your first suggestion results in mac layout like `Option+L = @` with dead keys, though the second suggestion results in an error

![2023-05-29T07:40:50,219258901+02:00](https://github.com/hyprwm/Hyprland/assets/25435034/e83ee907-318d-41ac-b965-f20c2b40758f)

It appears to error in `latin1` because just 
```
  kb_layout = de,de
  kb_variant = mac,nodeadkeys
```

results in no error but still dead, or rather swapped, keys though.

This is what I have so far.

```
input {
  kb_layout = de
  follow_mouse = 1
  sensitivity = 0 # -1.0 - 1.0, 0 means no modification.
}

device:logitech-mx-keys-mac {
  kb_layout = de,de
  kb_variant = mac,nodeadkeys
}
```

The wiki really needs more examples and remove the confusing ambiguity like this
```
If you want a single variant ie. pl/dvorak on one layout but us/qwerty on the other, xkb parameters can still be blank, however the amount of comma-separated parameters have to match. Alternatively, a single parameter can be specified for it to apply to all three.
```

How? If they have to match in count, and only one is passed, how can one be specified to match all three when the example says that the second one will be us/querty and not us/dvorak. The wiki is contradicting itself and needs more proper examples like how to split values of `localectl list-keymaps` into values for `kb_variant`

As a developer myself this is one of the worst and confusing wiki I've ever seen.
